File: android/src/main/java/top/huic/tencent_im_plugin/message/entity/FileMessageEntity.java
package top.huic.tencent_im_plugin.message.entity;

import com.tencent.imsdk.v2.V2TIMFileElem;

import top.huic.tencent_im_plugin.enums.MessageNodeType;

/**
 * 文件消息实体
 */
public class FileMessageEntity extends AbstractMessageEntity {
    /**
     * 文件路径
     */
    private String filePath;

    /**
     * 文件名
     */
    private String fileName;

    /**
     * 文件UUID
     */
    private String uuid;

    /**
     * 文件大小
     */
    private int size;

    public FileMessageEntity() {
        super(MessageNodeType.File);
    }

    public FileMessageEntity(V2TIMFileElem elem) {
        super(MessageNodeType.File);
        this.setFileName(elem.getFileName());
        this.setFilePath(elem.getPath());
        this.setSize(elem.getFileSize());
        this.setUuid(elem.getUUID());
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }
}

File: ios/Classes/message/entity/FileMessageEntity.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK

/// 文件消息实体
class FileMessageEntity: AbstractMessageEntity {
    /**
     * 文件路径
     */
    var filePath: String?;

    /**
     * 文件名
     */
    var fileName: String?;

    /**
     * 文件UUID
     */
    var uuid: String?;

    /**
     * 文件大小
     */
    var size: Int32?;

    override init() {
        super.init(MessageNodeType.File);
    }

    init(elem: V2TIMFileElem) {
        super.init(MessageNodeType.File);
        self.uuid = elem.uuid;
        self.fileName = elem.filename;
        self.filePath = elem.path;
        self.size = elem.fileSize;
    }
}
